work-tracker-servlet/src/main/java/com/deere/isg/worktracker/servlet/AbstractHttpWorkFilter.java

```diff
@@ -42,9 +42,9 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
         W payload = createWork(request);
         HttpServletRequest httpRequest = getHttpRequest(payload, request);
         OutstandingWork<W> outstanding = getOutstanding();
-
         try {
             if (outstanding != null) {
+                preProcess(request, response, payload);
```

Comment:
Why does this need done before going into outstanding? If it takes a while, we don't have any tracking for that time.

Reply:
It can be done outside of Outstanding. Do we want the preprocess to happen before we even create the work though (because the time starts when we create the work and that also means that the client will not have access to the payload at preprocess)?

Reply:
The clock need to start ASAP.  If we preprocess before it starts we start lying to ourselves.

In the case that drove this, does the preprocess need access to the Work, or does it specifically need to work before the Work is created?

Reply:
Could an app just use another Filter?  Then it is explicit to them that they are putting something before the WorkFilter.